mm2src/coins/rpc_command/account_balance.rs

```diff
@@ -26,48 +26,68 @@ pub struct AccountBalanceParams {
 }
 
 #[derive(Debug, PartialEq, Serialize)]
-pub struct HDAccountBalanceResponse {
+pub struct HDAccountBalanceResponse<BalanceMap> {
     pub account_index: u32,
     pub derivation_path: RpcDerivationPath,
-    pub addresses: Vec<HDAddressBalance>,
+    pub addresses: Vec<HDAddressBalance<BalanceMap>>,
     // Todo: Add option to get total balance of all addresses in addition to page_balance
-    pub page_balance: CoinBalance,
+    pub page_balance: BalanceMap,
     pub limit: usize,
     pub skipped: u32,
     pub total: u32,
     pub total_pages: usize,
     pub paging_options: PagingOptionsEnum<u32>,
 }
 
+/// Enum for the response of the `account_balance` RPC command.
+#[derive(Debug, PartialEq, Serialize)]
+#[serde(untagged)]
+pub enum HDAccountBalanceResponseEnum {
+    Single(HDAccountBalanceResponse<CoinBalance>),
+    Map(HDAccountBalanceResponse<CoinBalanceMap>),
+}
+
+/// Trait for the `account_balance` RPC command.
 #[async_trait]
 pub trait AccountBalanceRpcOps {
+    type BalanceMap;
+
     async fn account_balance_rpc(
         &self,
         params: AccountBalanceParams,
-    ) -> MmResult<HDAccountBalanceResponse, HDAccountBalanceRpcError>;
+    ) -> MmResult<HDAccountBalanceResponse<Self::BalanceMap>, HDAccountBalanceRpcError>;
 }
 
+/// `account_balance` RPC command implementation.
 pub async fn account_balance(
     ctx: MmArc,
     req: HDAccountBalanceRequest,
-) -> MmResult<HDAccountBalanceResponse, HDAccountBalanceRpcError> {
+) -> MmResult<HDAccountBalanceResponseEnum, HDAccountBalanceRpcError> {
     match lp_coinfind_or_err(&ctx, &req.coin).await? {
-        MmCoinEnum::UtxoCoin(utxo) => utxo.account_balance_rpc(req.params).await,
-        MmCoinEnum::QtumCoin(qtum) => qtum.account_balance_rpc(req.params).await,
+        MmCoinEnum::UtxoCoin(utxo) => Ok(HDAccountBalanceResponseEnum::Single(
+            utxo.account_balance_rpc(req.params).await?,
+        )),
+        MmCoinEnum::QtumCoin(qtum) => Ok(HDAccountBalanceResponseEnum::Single(
+            qtum.account_balance_rpc(req.params).await?,
+        )),
+        MmCoinEnum::EthCoin(eth) => Ok(HDAccountBalanceResponseEnum::Map(
+            eth.account_balance_rpc(req.params).await?,
```

Comment:
I also was going to ask: maybe use the same format for all coins, like for utxo just return a map of a single element, to simplify processing for the GUI ?

Reply:
> Could you please clarify why we use HDAccountBalanceResponseEnum::Map for the eth coin, and Single for the other supported assets?

This is a map of coins/tokens ticker to balance, so we use map for ETH/EVM because the response should contain for each address the platform coin balance and the tokens balances. Other coins don't have tokens currently, we can add tokens support for them in the future but we would use a new API method, for backward compatibility of current APIs we shouldn't return a map as this is used currently in GUIs (e.g. for UTXO trezor).

P.S. GUIs would need to change their code for already implemented functionalities if we used Map for previously implemented coins. It's not only GUIs but any one running the defi framework and uses these methods.

Reply:
> maybe use the same format for all coins

I want to do that in the future, but like I said they have to be new APIs.

Reply:
>  we use map for ETH/EVM because the response should contain for each address the platform coin balance and the tokens balances. Other coins don't have tokens currently

Oh tokens, exactly

Please add doc comments for `pub enum HDAccountBalanceResponseEnum` and for `pub struct HDAccountBalanceResponse<BalanceObject>` about this